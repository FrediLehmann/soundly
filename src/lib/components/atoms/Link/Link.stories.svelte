<script>
  import { Meta, Story, Template } from '@storybook/addon-svelte-csf';

  import Link from './Link.svelte';
  import { Settings } from '../../../Icons';
</script>

<Meta
  title="Atoms/Link"
  component={Link}
  argTypes={{
    href: {
      control: { type: 'text' },
      table: {
        type: {
          summary: 'Location to link to'
        }
      }
    },
    style: {
      control: { type: 'select' },
      options: ['none', 'default', 'primary', 'secondary', 'ghost', 'link'],
      table: {
        defaultValue: {
          summary: 'default'
        },
        type: {
          summary: 'The style of the Link'
        }
      }
    }
  }} />

<Template let:args>
  <div class="box">
    <Link {...args} on:click={e => e.preventDefault()}>Link</Link>
    <Link
      {...args}
      on:click={e => e.preventDefault()}
      href="https://storybook.js.org/">
      Visited Link
    </Link>
    <Link {...args} on:click={e => e.preventDefault()} aria-label="settings">
      <Settings />
    </Link>
    <Link {...args} on:click={e => e.preventDefault()}>
      Button
      <Settings />
    </Link>
    <Link {...args} on:click={e => e.preventDefault()}>
      <Settings />
      Button
    </Link>
  </div>
</Template>

<Story name="Default" args={{ style: 'default' }} />

<style>
  .box {
    display: flex;
    flex-direction: column;
    gap: 12px;
  }
</style>
