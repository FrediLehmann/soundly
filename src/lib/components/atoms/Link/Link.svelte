<script type="ts">
  import type { ButtonStyles } from '$lib/components/atoms';

  import { LinkStyles } from './LinkStyles.enum';

  export let href: string;
  export let style: LinkStyles | ButtonStyles = LinkStyles.default;
</script>

<a {href} data-type={style} class="link" {...$$restProps}>
  <slot />
</a>
