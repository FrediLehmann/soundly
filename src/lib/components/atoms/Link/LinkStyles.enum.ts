export enum LinkStyles {
  default = 'block text-blue-700',
  buttonPrimary = 'flex items-center px-5 py-2 h-full bg-blue-600 text-white',
  buttonSecondary = 'flex items-center px-5 py-2 h-full bg-gray-300'
}
