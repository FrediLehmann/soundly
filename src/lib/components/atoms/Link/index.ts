import Link from './Link.svelte';
import { LinkStyles } from './LinkStyles.enum';

export { Link, LinkStyles };
