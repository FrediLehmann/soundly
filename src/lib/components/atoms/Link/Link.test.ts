/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Link from './Link.svelte';
import { LinkStyles } from './LinkStyles.enum';

describe('<Link /> component', () => {
  test('render default Link', () => {
    const { getByRole } = render(Link, { href: '/test' });

    expect(getByRole('link')).toBeInTheDocument();
    expect(getByRole('link').dataset['type']).toEqual(LinkStyles.default);
    expect(getByRole('link').getAttribute('href')).toEqual('/test');
  });

  test('render correct styles', () => {
    for (const style in LinkStyles) {
      const rendered = render(Link, {
        href: '/test',
        style: style
      });

      expect(rendered.getByRole('link').dataset['type']).toEqual(style);
      rendered.unmount();
    }
  });

  test('custom class', () => {
    const { getByRole } = render(Link, { href: '/', class: 'added-class' });
    expect(getByRole('link').className).toEqual(`added-class link`);
  });
});
