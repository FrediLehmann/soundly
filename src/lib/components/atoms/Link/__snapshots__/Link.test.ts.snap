// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> component render Link 1`] = `
Link {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "/test",
      Object {},
      undefined,
      Object {},
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "href": 0,
    },
    "root": <div>
      <a
        class="svelte-ggsnh4"
        href="/test"
      />
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
